spec.html

```diff
@@ -819,8 +819,19 @@ <h1>[empty]</h1>
 
       <emu-clause id="sec-lookahead-restrictions">
         <h1>Lookahead Restrictions</h1>
-        <p>If the phrase “[lookahead = _seq_]” appears in the right-hand side of a production, it indicates that the production may only be used if the token sequence _seq_ is a prefix of the immediately following input token sequence. Similarly, “[lookahead ∈ _set_]”, where _set_ is a finite non-empty set of token sequences, indicates that the production may only be used if some element of _set_ is a prefix of the immediately following token sequence. For convenience, the set can also be written as a nonterminal, in which case it represents the set of all token sequences to which that nonterminal could expand. It is considered an editorial error if the nonterminal could expand to infinitely many distinct token sequences.</p>
-        <p>These conditions may be negated. “[lookahead ≠ _seq_]” indicates that the containing production may only be used if _seq_ is <em>not</em> a prefix of the immediately following input token sequence, and “[lookahead ∉ _set_]” indicates that the production may only be used if <em>no</em> element of _set_ is a prefix of the immediately following token sequence.</p>
+        <p>When a phrase of the form “[lookahead …]” appears in the right-hand side of a production, it indicates that the production may only be used if the “lookahead” (the items that immediately follow the corresponding point in the input) satisfies a specified constraint. If the production is in the syntactic grammar, the items of the lookahead are input elements (mainly tokens); otherwise, they are code points.</p>
+        <p>The forms of lookahead restriction, along with the constraint that each imposes on the lookahead, are as follows:</p>
+        <ul>
+          <li>“[lookahead = _seq_]”: _seq_ matches a prefix of the lookahead</li>
+          <li>“[lookahead ≠ _seq_]”: _seq_ does <em>not</em> match any prefix of the lookahead</li>
+          <li>“[lookahead ∈ _set_]”: some element of _set_ matches a prefix of the lookahead</li>
+          <li>“[lookahead ∉ _set_]”: <em>no</em> element of _set_ matches any prefix of the lookahead</li>
+        </ul>
+        <p>In the above:</p>
+        <ul>
+          <li>_seq_ is a sequence of terminal symbols from the production's grammar; and</li>
+          <li>_set_ is a finite non-empty set of terminal sequences. For convenience, _set_ can also be written as a nonterminal from the production's grammar, in which case it represents the set of all terminal sequences to which that nonterminal could expand. It is considered an editorial error if the nonterminal could expand to infinitely many distinct terminal sequences.</li>
```

Comment:
`seq` and `set` look too similar. I'd rename `seq` to `sequence`.

Reply:
The third commit merges them into `_pattern_`.